src/sentry/integrations/vsts/notify_action.py

```diff
@@ -1,42 +1,99 @@
-    # TODO 2.0 Add form fields.
+    vsts_integration = forms.ChoiceField(choices=(), widget=forms.Select())
+
     def __init__(self, *args, **kwargs):
+        integrations = [(i.id, i.name) for i in kwargs.pop("integrations")]
         super(AzureDevopsNotifyServiceForm, self).__init__(*args, **kwargs)
 
+        if integrations:
+            self.fields["vsts_integration"].initial = integrations[0][0]
+
+        self.fields["vsts_integration"].choices = integrations
+        self.fields["vsts_integration"].widget.choices = self.fields["vsts_integration"].choices
+
     def clean(self):
         return super(AzureDevopsNotifyServiceForm, self).clean()
 
 
 class AzureDevopsCreateTicketAction(TicketEventAction):
     form_cls = AzureDevopsNotifyServiceForm
-    label = u"TODO Create an Azure DevOps work item in {name} with these "
+
     ticket_type = "an Azure DevOps work item"
     link = "https://docs.sentry.io/product/integrations/azure-devops/#issue-sync"
+    label = u"Create an Azure DevOps work item in {vsts_integration} with these "
+    prompt = "Create an Azure DevOps work item"
     provider = "vsts"
     integration_key = INTEGRATION_KEY
 
     def __init__(self, *args, **kwargs):
         super(AzureDevopsCreateTicketAction, self).__init__(*args, **kwargs)
-        # TODO 2.1 Add form_fields
-        self.form_fields = {}
+        integration_choices = [(i.id, i.name) for i in self.get_integrations()]
+
+        if not self.get_integration_id() and integration_choices:
+            self.data[self.integration_key] = integration_choices[0][0]
+
+        self.form_fields = {
+            "vsts_integration": {
+                "choices": integration_choices,
+                "initial": six.text_type(self.get_integration_id()),
+                "type": "choice",
+                "updatesForm": True,
+            },
+        }
+
+        dynamic_fields = self.get_dynamic_form_fields()
+        if dynamic_fields:
+            for field in dynamic_fields:
+                self.form_fields[field["name"]] = field
 
     def render_label(self):
-        return self.label.format(name=self.get_integration_name())
+        return self.label.format(vsts_integration=self.get_integration_name())
+
+    def get_dynamic_form_fields(self):
+        """
+        Either get the dynamic form fields cached on the DB or make an API call
+        to VSTS to get them for the selected integration. If both fail, return `None`.
+
+        :return: Django form fields dictionary
+        """
+        if "dynamic_form_fields" in self.data:
+            return self.data["dynamic_form_fields"]
+
+        try:
+            integration = self.get_integration()
+        except Integration.DoesNotExist:
+            pass
+        else:
+            vsts_integration = VstsIntegration(integration, self.project.organization.id)
+            try:
+                fields = vsts_integration.get_create_issue_config_no_args()
+            except IntegrationError as e:
```

Comment:
@ceorourke I realize we do the same thing in Jira, but this is going to cause a weird bug in the UI when the API fails. We need to return an error status when the API fails here and let the client handle it.